share/sysbuild/CMakeLists.txt

```diff
@@ -18,7 +18,7 @@ set(APP_DIR ${APP_DIR} CACHE PATH "Main Application Source Directory")
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake/modules)
 # List of Zephyr and sysbuild CMake modules we need for sysbuild.
 # Note: sysbuild_kconfig will internally load kconfig CMake module.
-set(zephyr_modules extensions sysbuild_extensions python west root zephyr_module boards shields sysbuild_kconfig)
+set(zephyr_modules extensions sysbuild_extensions python west unittest_root root zephyr_module boards shields sysbuild_kconfig)
```

Comment:
I don't like this, this means that sysbuild can always find the unit testing board, which it should not be able to do unless running a unit test

Reply:
Sysbuild doesn't know whether it's running a unit test. It's up to the image itself to load the Zephyr package in unittest mode, which then determines which boards can be selected. Hence, I figured sysbuild should always find the board and let individual images fail if the board is invalid. Effectively, sysbuild can include both unit tests and regular applications at any time.

Reply:
> Effectively, sysbuild can include both unit tests and regular applications at any time.

In my opinion it should not. Zephyr applications do not, and sysbuild before this PR did not.

Reply:
Yes, Zephyr applications are either/or, but I don't see a reason why sysbuild, as a higher-order build system, must be limited to either/or. Or why it shouldn't support unit tests at all.

Reply:
Because this makes sysbuild accept that the unit testing board is valid, there's nothing wrong with allowing sysbuild to use the unit testing board, but it should only be found as valid when sysbuild is being used in unit testing mode, it shouldn't have to configure an image before throwing an error that the board provided is not valid

Reply:
i'm aligned with @nordicjm on this.

Secondly, why do you need to invoke sysbuild in order to test the functionality of `zephyr_get()` ?
Your test case can just create the `<image>_sysbuild_cache.txt` file and populate it with the settings you want to test, and thereafter call the `zephyr_get()` to see that it behaves as expected.
No reason to involves sysbuild in those tests, especially not as you are not testing sysbuild itself, you're just testing `zephyr_get()`.

Alternatively you may use the sysbuild extensions to create the image build directly, for `zephyr_get()` testing:
```
list(APPEND CMAKE_MODULE_PATH ${ZEPHYR_BASE}/share/sysbuild/cmake/modules)
include(sysbuild_extensions)

ExternalZephyrProject_Add(....)
ExternalZephyrProject_Cmake(....)
```
for testing that the sysbuild extension command correct creates a proper sysbuyild cache file.

Bote the two presented approaches above remove the need for sysbuild to know about the `unit_testing` board.
And the `unit_testing` board can still be used to speed up the tests as you present with this PR, cause all tests are done as a regular Zephyr image build.

Reply:
> Secondly, why do you need to invoke sysbuild in order to test the functionality of `zephyr_get()` ?

I wish you had made that point before approving #61360.

Testing with sysbuild is easier, but by no means necessary. Mocking the sysbuild cache is always an option.